install_deps.sh

```diff
@@ -401,51 +400,70 @@ wget ${WGET_FLAGS} https://archive.apache.org/dist/thrift/0.11.0/thrift-0.11.0.t
- tar -xvzf cppcheck.tar.gz && rm ./cppcheck.tar.gz  && \
- cd cppcheck-${CPPCHECK_VER} && \
- mkdir build && cd build && \
- cmake .. && \
- cmake --build . && \
- make install && \
- cd ${HOME} && \
- rm -rf cppcheck-${CPPCHECK_VER}
-
-# Jaeger really wants to find BoostConfig.cmake, not FindBoost.cmake.
-# This wasn't introduced until boost 1.70.
-# Jaegertracing.cmake finds FindBoost.cmake first anyways.
-# The following sed just removes the search for BoostConfig.cmake.
-sed -i '/boost_components/d' /usr/local/lib/cmake/jaegertracing/jaegertracingConfig.cmake
+  # TODO: Upgrade to opentelemetry-cpp
+  # Tracing via Jaeger and Thrift protocol
+  # Copy FindThrift.cmake because installing Thrift does not include a CMake definition
+  cd $HOME
+  git clone -b v0.7.0 --depth 1 https://github.com/jaegertracing/jaeger-client-cpp && \
+      cd jaeger-client-cpp && \
+      mkdir build && \
+      cd build && \
+      cmake -DHUNTER_ENABLED=OFF -DHUNTER_BUILD_SHARED_LIBS=OFF -DBUILD_TESTING=OFF \
+            -DBUILD_SHARED_LIBS=OFF -DJAEGERTRACING_BUILD_EXAMPLES=OFF \
+            -DJAEGERTRACING_PLUGIN=OFF -DJAEGERTRACING_COVERAGE=OFF \
+            -DJAEGERTRACING_BUILD_CROSSDOCK=OFF -DJAEGERTRACING_WITH_YAML_CPP=OFF \
+            .. && \
+      make -j$(nproc) install && \
+      cp ../cmake/Findthrift.cmake /usr/share/cmake-3.20/Modules/ && \
+      cd ${HOME} && \
+      rm -r jaeger-client-cpp
 
+  # Jaeger really wants to find BoostConfig.cmake, not FindBoost.cmake.
+  # This wasn't introduced until boost 1.70.
+  # Jaegertracing.cmake finds FindBoost.cmake first anyways.
+  # The following sed just removes the search for BoostConfig.cmake.
+  sed -i '/boost_components/d' /usr/local/lib/cmake/jaegertracing/jaegertracingConfig.cmake
 }
 
+install_cppcheck(){
+  cd ${HOME}
+  CPPCHECK_VER="2.8"
+  wget ${WGET_FLAGS} https://sourceforge.net/projects/cppcheck/files/cppcheck/${CPPCHECK_VER}/cppcheck-${CPPCHECK_VER}.tar.gz/download -O ./cppcheck.tar.gz && \
+  tar -xvzf cppcheck.tar.gz && rm ./cppcheck.tar.gz  && \
+  cd cppcheck-${CPPCHECK_VER} && \
+  mkdir build && cd build && \
+  cmake .. && \
+  cmake --build . && \
+  make install && \
+  cd ${HOME} && \
+  rm -rf cppcheck-${CPPCHECK_VER}
+}
 
 
+install_ccache(){
+  # ccache is used to accelerate C/C++ recompilation
+  CCACHE_VER=4.6.1
+  cd "$HOME"
+  wget https://github.com/ccache/ccache/releases/download/v${CCACHE_VER}/ccache-${CCACHE_VER}.tar.xz && \
+    tar xvf ccache-${CCACHE_VER}.tar.xz && \
+    cd ccache-${CCACHE_VER} && \
+    mkdir build && cd build && \
+    cmake -DCMAKE_BUILD_TYPE=Release -DREDIS_STORAGE_BACKEND=OFF .. && \
+    make && \
+    make install && \
+    cd "${HOME}" && \
+    rm -rf ccache-${CCACHE_VER} ccache-${CCACHE_VER}.tar.xz
+  CCACHE_PATH=$(which ccache)
+  ln -s "${CCACHE_PATH}" /usr/local/bin/gcc
```

Comment:
why make this symbolic links?
the compiler command should look like:
`ccache /usr/local/bin/clang++ ...` making it possible to choose between enabling/disabling ccache usage and choosing a compiler

Reply:
There are 2 ways to run ccache and the symbolic link is a valid one. ccache knows how to find the compiler according to the nvoked sym link name, so there is no way it will not run the right g++/clang++.
To simplify native build, It's better to have it without sym links. I'll check if it's possible to run it the way you suggest..

Reply:
We don't need to use ccache explcitly. I checked that. the CMake CMAKE_CXX_COMPILER_LAUNCHER definition do the trick, all we need to do is to pass it as CMake definition when building